provider/server/machine_list_test.go

```diff
@@ -13,28 +13,38 @@ import (
 )
 
 var _ = Describe("ListMachine", func() {
-	It("should correctly list machines", func(ctx SpecContext) {
+	It("should list machines", func(ctx SpecContext) {
 		By("creating a machine")
 		createResp, err := machineClient.CreateMachine(ctx, &iri.CreateMachineRequest{
 			Machine: &iri.Machine{
 				Metadata: &irimeta.ObjectMetadata{
 					Labels: map[string]string{
-						"machinepoolletv1alpha1.MachineUIDLabel": "foobar",
+						"foo": "bar",
 					},
 				},
 				Spec: &iri.MachineSpec{
 					Power: iri.Power_POWER_ON,
 					Class: machineClassx3xlarge,
+					Volumes: []*iri.Volume{
+						{
+							Name: "disk-1",
+							EmptyDisk: &iri.EmptyDisk{
+								SizeBytes: 5368709120,
+							},
+							Device: "oda",
+						},
+					},
+					NetworkInterfaces: nil,
 				},
 			},
 		})
 		Expect(err).NotTo(HaveOccurred())
 		Expect(createResp).NotTo(BeNil())
 
 		DeferCleanup(func(ctx SpecContext) {
-			_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
-			Expect(err).ShouldNot(HaveOccurred())
 			Eventually(func() bool {
+				_, err := machineClient.DeleteMachine(ctx, &iri.DeleteMachineRequest{MachineId: createResp.Machine.Metadata.Id})
```

Comment:
Any specific reason for shifting this inside the Eventually block?

Reply:
No,  have taken it out of Eventually block